starknet-replay/src/storage/rpc/rpc_client.rs

```diff
@@ -0,0 +1,429 @@
+            .await?;
+        match block_header {
+            MaybePendingBlockWithTxHashes::Block(block) => {
+                let sequencer: StarkHash =
+                    Felt::from_bytes_be(&block.sequencer_address.to_bytes_be());
+                let price_in_fri: u128 = block.l1_gas_price.price_in_fri.to_string().parse()?;
+                let price_in_wei: u128 = block.l1_gas_price.price_in_wei.to_string().parse()?;
+
+                let data_price_in_fri: u128 =
+                    block.l1_data_gas_price.price_in_fri.to_string().parse()?;
+                let data_price_in_wei: u128 =
+                    block.l1_data_gas_price.price_in_wei.to_string().parse()?;
+
+                let block_header: BlockHeader = BlockHeader {
+                    block_hash: BlockHash(Felt::from_bytes_be(&block.block_hash.to_bytes_be())),
+                    parent_hash: BlockHash(Felt::from_bytes_be(&block.parent_hash.to_bytes_be())),
+                    block_number: starknet_api::block::BlockNumber(block.block_number),
+                    l1_gas_price: GasPricePerToken {
+                        price_in_fri: GasPrice(price_in_fri),
+                        price_in_wei: GasPrice(price_in_wei),
+                    },
+                    l1_data_gas_price: GasPricePerToken {
+                        price_in_fri: GasPrice(data_price_in_fri),
+                        price_in_wei: GasPrice(data_price_in_wei),
+                    },
+                    state_root: GlobalRoot(Felt::from_bytes_be(&block.new_root.to_bytes_be())),
+                    sequencer: SequencerContractAddress(sequencer.try_into()?),
+                    timestamp: BlockTimestamp(block.timestamp),
+                    l1_da_mode: match block.l1_da_mode {
+                        starknet_core::types::L1DataAvailabilityMode::Blob => {
+                            L1DataAvailabilityMode::Blob
+                        }
+                        starknet_core::types::L1DataAvailabilityMode::Calldata => {
+                            L1DataAvailabilityMode::Calldata
+                        }
+                    },
+                    state_diff_commitment: None,
+                    transaction_commitment: None,
+                    event_commitment: None,
+                    n_transactions: block.transactions.len(),
+                    n_events: 0,
+                    starknet_version: StarknetVersion(block.starknet_version),
+                    state_diff_length: None,
+                    receipt_commitment: None,
+                };
+                Ok(block_header)
+            }
+            MaybePendingBlockWithTxHashes::PendingBlock(_) => unreachable!(),
+        }
+    }
+
+    /// This function queries the transactions and receipts in a block.
+    ///
+    /// # Arguments
+    ///
+    /// - `block_number`: the block number to be queried.
+    ///
+    /// # Errors
+    ///
+    /// Returns [`Err`] if the request fails or the block number doesn't exist.
```

Comment:
Same comment as above.

Reply:
This method is complex, it can return different types of error.